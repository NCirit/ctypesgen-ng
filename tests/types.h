#include <complex.h>
#include <stdbool.h>
#include <stddef.h>
#include <stdint.h>
#include <uchar.h>
#include <sys/types.h>

void int_types(signed char b, short s, int i, long l, long long ll);

void intx_types(int8_t b, int16_t s, int32_t l, int64_t ll);

void uint_types(unsigned char b, unsigned short s, unsigned int i, unsigned long l, unsigned long long ll);

void uintx_types(uint8_t b, uint16_t s, uint32_t l, uint64_t ll);

void float_types(float f, double d, long double ld);

void char_types(char c, char *s, char16_t c16, char16_t *s16, char32_t c32, char32_t *s32);

void wchar_types(wchar_t wc, wchar_t *wcs);

void bool_types(_Bool b, bool b2);

void complex_types(float _Complex fc, double _Complex dc, long double _Complex ldc);

void complex_types2(float complex fc, double complex dc, long double complex ldc);

void big_int_types(__int128 i, unsigned __int128 u);

void big_int_typedefs(__int128_t i,  __uint128_t u);

void size_types(size_t s, ssize_t ss); 
