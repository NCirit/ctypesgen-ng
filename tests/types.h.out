import ctypes
import ctypes.util
import os


if not ('LIBRARY_PATH' in os.environ):
    if 'PYTHONPATH' in os.environ:
        os.environ['LIBRARY_PATH'] = os.environ['PYTHONPATH']
    else:
        os.environ['LIBRARY_PATH'] = os.path.dirname(os.path.abspath(__file__))

if os.name == "nt":
    if 'PYTHONPATH' in os.environ:
        os.environ['PATH'] += os.pathsep + os.environ['PYTHONPATH']
    else:
        os.environ['PATH'] += os.pathsep + os.path.dirname(os.path.abspath(__file__))
test_path = ctypes.util.find_library('test')
test = ctypes.CDLL(os.path.abspath(test_path))

test.int_types.restype = None
test.int_types.argtypes = [ctypes.c_byte, ctypes.c_short, ctypes.c_int, ctypes.c_long, ctypes.c_longlong]

test.intx_types.restype = None
test.intx_types.argtypes = [ctypes.c_int8, ctypes.c_int16, ctypes.c_int32, ctypes.c_int64]

test.uint_types.restype = None
test.uint_types.argtypes = [ctypes.c_ubyte, ctypes.c_ushort, ctypes.c_uint, ctypes.c_ulong, ctypes.c_ulonglong]

test.uintx_types.restype = None
test.uintx_types.argtypes = [ctypes.c_uint8, ctypes.c_uint16, ctypes.c_uint32, ctypes.c_uint64]

test.float_types.restype = None
test.float_types.argtypes = [ctypes.c_float, ctypes.c_double, ctypes.c_longdouble]

test.char_types.restype = None
test.char_types.argtypes = [ctypes.c_char, ctypes.c_char_p, typedef_char16_t, ctypes.POINTER(typedef_char16_t), typedef_char32_t, ctypes.POINTER(typedef_char32_t)]

test.wchar_types.restype = None
test.wchar_types.argtypes = [ctypes.c_wchar, ctypes.POINTER(ctypes.c_wchar)]

test.bool_types.restype = None
test.bool_types.argtypes = [ctypes.c_bool, ctypes.c_bool]

test.complex_types.restype = None
test.complex_types.argtypes = [(ctypes.c_float * 2), (ctypes.c_double * 2), (ctypes.c_longdouble * 2)]

test.complex_types2.restype = None
test.complex_types2.argtypes = [(ctypes.c_float * 2), (ctypes.c_double * 2), (ctypes.c_longdouble * 2)]

test.big_int_types.restype = None
test.big_int_types.argtypes = [(ctypes.c_ubyte * 16), (ctypes.c_ubyte * 16)]

test.big_int_typedefs.restype = None
test.big_int_typedefs.argtypes = [typedef___int128_t, typedef___uint128_t]

test.size_types.restype = None
test.size_types.argtypes = [ctypes.c_size_t, ctypes.c_ssize_t]
